stablehlo/tests/transforms/stablehlo_legalize_qdq_to_quantized_op.mlir

```diff
@@ -0,0 +1,32 @@
+// RUN: stablehlo-opt %s -verify-diagnostics -split-input-file -allow-unregistered-dialect --stablehlo-legalize-qdq-to-quantized-op | FileCheck %s --check-prefixes=CHECK
+
+// -----
+
+// CHECK-LABEL @compose_quantized_abs_op
+// CHECK: %[[abs0:.*]] = stablehlo.abs %arg0 : tensor<16x16x!quant.uniform<u8:f32, 3.400000e+01:16>>
+// CHECK-NEXT: %[[abs1:.*]] = stablehlo.abs %[[abs0]] : tensor<16x16x!quant.uniform<u8:f32, 3.400000e+01:16>>
+// CHECK-NEXT: return %[[abs1]] : tensor<16x16x!quant.uniform<u8:f32, 3.400000e+01:16>>
+func.func @compose_quantized_abs_op(%arg0: tensor<16x16x!quant.uniform<ui8:f32, 34.0:16>>) -> tensor<16x16x!quant.uniform<ui8:f32, 34.0:16>> {
+    %0 = stablehlo.uniform_dequantize %arg0 : (tensor<16x16x!quant.uniform<ui8:f32, 34.0:16>>) -> tensor<16x16xf32>
+    %1 = stablehlo.abs %0 : tensor<16x16xf32>
+    %2 = stablehlo.uniform_quantize %1 : (tensor<16x16xf32>) -> tensor<16x16x!quant.uniform<ui8:f32, 34.0:16>>
+    %3 = stablehlo.abs %2 : tensor<16x16x!quant.uniform<ui8:f32, 34.0:16>>
```

Comment:
You can keep the first test to showcase the functioning of the pass., just with `%0, %1, %2, return %2`. No need for the trailing quantized abs. In separate tests you can showcase a case the pass cannot successfully pattern match: These will be mostly the cases we return `notifyMatchFailure` (e.g. hybrid case, ops with regions etc.).

Reply:
we can use the current testfile and write a test which the pattern would failed to match and then check the resulting program (which will be mostly unconverted) using check directive.